app/src/main/res/layout/list_item_trip.xml

```diff
@@ -87,24 +87,41 @@
 			android:layout_marginRight="8dp"
 			android:layout_marginTop="4dp"
 			app:flexWrap="wrap"
-			app:layout_constraintEnd_toStartOf="@+id/duration"
+			app:layout_constraintEnd_toStartOf="@+id/durationPrice"
 			app:layout_constraintStart_toStartOf="@+id/fromLocation"
 			app:layout_constraintTop_toBottomOf="@+id/fromLocation">
 
 			<include layout="@layout/list_item_line"/>
 
 		</com.google.android.flexbox.FlexboxLayout>
 
-		<TextView
-			android:id="@+id/duration"
+		<LinearLayout
+			android:id="@+id/durationPrice"
 			android:layout_width="wrap_content"
 			android:layout_height="wrap_content"
-			android:layout_marginBottom="8dp"
-			android:layout_marginTop="8dp"
-			app:layout_constraintBottom_toBottomOf="@+id/lines"
-			app:layout_constraintRight_toRightOf="parent"
+			android:orientation="vertical"
+			android:gravity="center_vertical"
+			app:layout_constraintEnd_toEndOf="parent"
 			app:layout_constraintTop_toTopOf="@+id/lines"
-			tools:text="1:30"/>
+			app:layout_constraintBottom_toBottomOf="@+id/lines">
```

Comment:
Would be nice to not introduce another hierarchy here, but use the existing ConstraintLayout for layouting.

Reply:
Here I've encountered the same problem as described above with layouting for both cases with and without price information.

Reply:
See above.